tests/test_dirs.py

```diff
@@ -0,0 +1,276 @@
+"""
+Copyright (c) 2021 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, Iterable
+
+import pytest
+
+from atomic_reactor.constants import DOCKERFILE_FILENAME, Platform
+from atomic_reactor.dirs import DockerfileNotExist, BuildDir, RootBuildDir, ImageInspectionData
+from atomic_reactor.source import DummySource
+from dockerfile_parse import DockerfileParser
+
+
+def test_builddir_failure_on_nonexisting_path():
+    with pytest.raises(FileNotFoundError, match="does not exist"):
+        BuildDir(Path("some_dir"), Platform.x86_64)
+
+
+def test_builddir_dockerfile_path(tmpdir):
+    dir_path = Path(tmpdir)
```

Comment:
You can use [tmp_path](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture) instead of `tmpdir` (completely optional, I'm fine with keeping it this way)